src/core/qgsmaplayer.cpp

```diff
@@ -1166,7 +1167,7 @@ bool QgsMapLayer::importNamedStyle( QDomDocument &myDocument, QString &myErrorMe
   }
 #endif
 
-  return readSymbology( myRoot, myErrorMessage );
+  return readSymbology( myRoot, myErrorMessage, QgsPathResolver() ); // TODO: support relative paths?
```

Comment:
Yes, please! Symbols stored relative to the project file would be very handy, providing a means of encapsulating sharable ZIP archives of projects, like with [QConsolidate](https://plugins.qgis.org/plugins/qconsolidate/). (same for `writeSymbology` TODO below).

This probably means that a standard-named symbols folder within the project folder, e.g. `symbols` is dynamically searched, if there is a project (or layer) setting to use project-relative symbol paths.